### ISSUE TITLE
Winget stops upon fail to install Microsoft Visual C++ 2013 Redistributable (x64)

### ISSUE THREAD

**Issue by birbilis** [microsoft/winget-cli#2259]
### Brief description of your issue

Doing a 
`winget upgrade --all -h`

I see winget exiting upon failure to install _Microsoft Visual C++ 2013 Redistributable (x64)_

> Found Driver Booster 9 [IObit.DriverBooster9] Version 9.3.0
> Found Product Portal [iZotope.ProductPortal] Version 1.4.3
> Found Unity [UnityTechnologies.Unity.2020] Version 2020.3.26f1
> Found WhySoSlow [Resplendence.WhySoSlow] Version 1.61
> Found Slack [SlackTechnologies.Slack] Version 4.27.154
> Found WinSCP [WinSCP.WinSCP] Version 5.21
> Found Microsoft Visual C++ 2013 Redistributable (x64) [Microsoft.VC++2013Redist-x64] Version 12.0.40664.0
> Found Eclipse Temurin JDK with Hotspot 8 [EclipseAdoptium.Temurin.8] Version 8.0.332.9
> Found LibreOffice [TheDocumentFoundation.LibreOffice] Version 7.3.4.2
> Found Microsoft XNA Framework Redistributable Refresh [Microsoft.XNARedist] Version 4.0.30901.0
> Found Microsoft Visual C++ 2015-2022 Redistributable (x86) [Microsoft.VC++2015-2022Redist-x86] Version 14.32.31326.0
> Found Microsoft Visual C++ 2013 Redistributable (x86) [Microsoft.VC++2013Redist-x86] Version 12.0.40664.0
> Found Poedit [VaclavSlavik.Poedit] Version 3.0
> Found GOG GALAXY [GOG.Galaxy] Version 2.0.49.1
> Found Microsoft Visual Studio Code [Microsoft.VisualStudioCode] Version 1.68.1
> Found Python 3 [Python.Python.3] Version 3.10.5150.0
> Found Tiled [Tiled.Tiled] Version 1.8.5
> Found Epic Games Launcher [EpicGames.EpicGamesLauncher] Version 1.3.23.0
> Found OpenVPN [OpenVPNTechnologies.OpenVPN] Version 2.5.036
> Found Unity Hub [UnityTechnologies.UnityHub] Version 3.1.2
> Found Viber [Viber.Viber] Version 17.8.1.0
> Found Waves Central [WavesAudio.WavesCentral] Version 13.1.4
> (1/22) Found Driver Booster 9 [IObit.DriverBooster9] Version 9.3.0
> This application is licensed to you by its owner.
> Microsoft is not responsible for, nor does it grant any licenses to, third-party packages.
> Downloading https://cdn.iobit.com/dl/driver_booster_setup.exe
>   ██████████████████████████████  24.5 MB / 24.5 MB
> Installer hash does not match; this cannot be overridden when running as admin
> 
> (2/22) Found Product Portal [iZotope.ProductPortal] Version 1.4.3
> This application is licensed to you by its owner.
> Microsoft is not responsible for, nor does it grant any licenses to, third-party packages.
> Downloading https://s3.amazonaws.com/izotopedownloads/product_download/iZotope_Product_Portal_v1_4_3.exe
>   ██████████████████████████████   113 MB /  113 MB
> Successfully verified installer hash
> Starting package install...
> Successfully installed
> 
> (3/22) Found Unity [UnityTechnologies.Unity.2020] Version 2020.3.26f1
> This application is licensed to you by its owner.
> Microsoft is not responsible for, nor does it grant any licenses to, third-party packages.
> Downloading https://download.unity3d.com/download_unity/7298b473bc1a/Windows64EditorInstaller/UnitySetup64-2020.3.26f1.exe
>   ██████████████████████████████  2.47 GB / 2.47 GB
> Successfully verified installer hash
> Starting package install...
> Successfully installed
> 
> (4/22) Found WhySoSlow [Resplendence.WhySoSlow] Version 1.61
> This application is licensed to you by its owner.
> Microsoft is not responsible for, nor does it grant any licenses to, third-party packages.
> Downloading https://resplendence.com/download/WhySoSlowSetup.exe
>   ██████████████████████████████  2.89 MB / 2.89 MB
> Successfully verified installer hash
> Starting package install...
> Successfully installed
> 
> (5/22) Found Slack [SlackTechnologies.Slack] Version 4.27.154
> This application is licensed to you by its owner.
> Microsoft is not responsible for, nor does it grant any licenses to, third-party packages.
> Downloading https://downloads.slack-edge.com/releases/windows/4.27.154/prod/x64/SlackSetup.exe
>   ██████████████████████████████  98.6 MB / 98.6 MB
> Successfully verified installer hash
> Starting package install...
> Successfully installed
> 
> (6/22) Found WinSCP [WinSCP.WinSCP] Version 5.21
> This application is licensed to you by its owner.
> Microsoft is not responsible for, nor does it grant any licenses to, third-party packages.
> Downloading https://sourceforge.net/projects/winscp/files/WinSCP/5.21/WinSCP-5.21-Setup.exe/download
>   ██████████████████████████████  10.9 MB / 10.9 MB
> Successfully verified installer hash
> Starting package install...
> Successfully installed
> 
> (7/22) Found Microsoft Visual C++ 2013 Redistributable (x64) [Microsoft.VC++2013Redist-x64] Version 12.0.40664.0
> This application is licensed to you by its owner.
> Microsoft is not responsible for, nor does it grant any licenses to, third-party packages.
> Downloading https://download.visualstudio.microsoft.com/download/pr/10912041/cee5d6bca2ddbcd039da727bf4acb48a/vcredist_x64.exe
>   ██████████████████████████████  6.86 MB / 6.86 MB
> Successfully verified installer hash
> Starting package uninstall...
> An unexpected error occurred while executing the command:
> 0x800401f5 : Application not found

### Steps to reproduce

probably machine-specific, try with 
Microsoft Visual C++ 2013 Redistributable (x64) [Microsoft.VC++2013Redist-x64] Version 12.0.40664.0

### Expected behavior

winget should continue with next installer (and maybe also not fail with that installer in the first place either)

### Actual behavior

winget stops  saying:

> An unexpected error occurred while executing the command:
0x800401f5 : Application not found

### Environment

```shell
winget --info
Windows Package Manager v1.2.10941
Copyright (c) Microsoft Corporation. All rights reserved.

Windows: Windows.Desktop v10.0.19044.1766
Package: Microsoft.DesktopAppInstaller v1.17.10941.0
```

**Comment by birbilis** [comment]
running a 2nd time seems to fetch and install again some packages that it had mentioned as installed, then fails again at same item with same error code and exits back to (admin) command prompt

> 
> Found Driver Booster 9 [IObit.DriverBooster9] Version 9.3.0
> Found Product Portal [iZotope.ProductPortal] Version 1.4.3
> Found Unity [UnityTechnologies.Unity.2020] Version 2020.3.26f1
> Found WhySoSlow [Resplendence.WhySoSlow] Version 1.61
> Found Microsoft Visual C++ 2013 Redistributable (x64) [Microsoft.VC++2013Redist-x64] Version 12.0.40664.0
> Found Eclipse Temurin JDK with Hotspot 8 [EclipseAdoptium.Temurin.8] Version 8.0.332.9
> Found LibreOffice [TheDocumentFoundation.LibreOffice] Version 7.3.4.2
> Found Microsoft XNA Framework Redistributable Refresh [Microsoft.XNARedist] Version 4.0.30901.0
> Found Microsoft Visual C++ 2015-2022 Redistributable (x86) [Microsoft.VC++2015-2022Redist-x86] Version 14.32.31326.0
> Found Microsoft Visual C++ 2013 Redistributable (x86) [Microsoft.VC++2013Redist-x86] Version 12.0.40664.0
> Found Poedit [VaclavSlavik.Poedit] Version 3.0
> Found GOG GALAXY [GOG.Galaxy] Version 2.0.49.1
> Found Microsoft Visual Studio Code [Microsoft.VisualStudioCode] Version 1.68.1
> Found Python 3 [Python.Python.3] Version 3.10.5150.0
> Found Tiled [Tiled.Tiled] Version 1.8.5
> Found Epic Games Launcher [EpicGames.EpicGamesLauncher] Version 1.3.23.0
> Found OpenVPN [OpenVPNTechnologies.OpenVPN] Version 2.5.036
> Found Unity Hub [UnityTechnologies.UnityHub] Version 3.1.2
> Found Viber [Viber.Viber] Version 17.8.1.0
> Found Waves Central [WavesAudio.WavesCentral] Version 13.1.4
> (1/20) Found Driver Booster 9 [IObit.DriverBooster9] Version 9.3.0
> This application is licensed to you by its owner.
> Microsoft is not responsible for, nor does it grant any licenses to, third-party packages.
> Downloading https://cdn.iobit.com/dl/driver_booster_setup.exe
>   ██████████████████████████████  24.5 MB / 24.5 MB
> Installer hash does not match; this cannot be overridden when running as admin
> 
> (2/20) Found Product Portal [iZotope.ProductPortal] Version 1.4.3
> This application is licensed to you by its owner.
> Microsoft is not responsible for, nor does it grant any licenses to, third-party packages.
> Downloading https://s3.amazonaws.com/izotopedownloads/product_download/iZotope_Product_Portal_v1_4_3.exe
>   ██████████████████████████████   113 MB /  113 MB
> Successfully verified installer hash
> Starting package install...
> Successfully installed
> 
> (3/20) Found Unity [UnityTechnologies.Unity.2020] Version 2020.3.26f1
> This application is licensed to you by its owner.
> Microsoft is not responsible for, nor does it grant any licenses to, third-party packages.
> Downloading https://download.unity3d.com/download_unity/7298b473bc1a/Windows64EditorInstaller/UnitySetup64-2020.3.26f1.exe
>   ██████████████████████████████  2.47 GB / 2.47 GB
> Successfully verified installer hash
> Starting package install...
> Successfully installed
> 
> (4/20) Found WhySoSlow [Resplendence.WhySoSlow] Version 1.61
> This application is licensed to you by its owner.
> Microsoft is not responsible for, nor does it grant any licenses to, third-party packages.
> Downloading https://resplendence.com/download/WhySoSlowSetup.exe
>   ██████████████████████████████  2.89 MB / 2.89 MB
> Successfully verified installer hash
> Starting package install...
> Successfully installed
> 
> (5/20) Found Microsoft Visual C++ 2013 Redistributable (x64) [Microsoft.VC++2013Redist-x64] Version 12.0.40664.0
> This application is licensed to you by its owner.
> Microsoft is not responsible for, nor does it grant any licenses to, third-party packages.
> Successfully verified installer hash
> Starting package uninstall...
> An unexpected error occurred while executing the command:
> 0x800401f5 : Application not found

**Comment by denelon** [comment]
Duplicate of #1442 

We've addressed this Issue in the 1.3 preview builds and in the 1.3 release candidate.

**Comment by birbilis** [comment]
another strange thing I noticed is that if I cancel (CTRL+C) packages download (btw, it would be nice to be able to press ESC to skip a package download and continue with the rest), next time I run it (in the same command prompt window, though I guess it shouldn't matter), I see those packages it had claimed to have installed, being downloaded/installed again

> 
> C:\WINDOWS\system32>winget upgrade --all -h
> Found 7-Zip [7zip.7zip] Version 22.00
> Found Driver Booster 9 [IObit.DriverBooster9] Version 9.3.0
> Found Product Portal [iZotope.ProductPortal] Version 1.4.3
> Found Unity [UnityTechnologies.Unity.2020] Version 2020.3.26f1
> Found WhySoSlow [Resplendence.WhySoSlow] Version 1.61
> Found Microsoft Visual C++ 2013 Redistributable (x64) [Microsoft.VC++2013Redist-x64] Version 12.0.40664.0
> Found Eclipse Temurin JDK with Hotspot 8 [EclipseAdoptium.Temurin.8] Version 8.0.332.9
> Found LibreOffice [TheDocumentFoundation.LibreOffice] Version 7.3.4.2
> Found Microsoft XNA Framework Redistributable Refresh [Microsoft.XNARedist] Version 4.0.30901.0
> Found Microsoft Visual C++ 2015-2022 Redistributable (x86) [Microsoft.VC++2015-2022Redist-x86] Version 14.32.31326.0
> Found Microsoft Visual C++ 2013 Redistributable (x86) [Microsoft.VC++2013Redist-x86] Version 12.0.40664.0
> Found Poedit [VaclavSlavik.Poedit] Version 3.0
> Found GOG GALAXY [GOG.Galaxy] Version 2.0.49.1
> Found Microsoft Visual Studio Code [Microsoft.VisualStudioCode] Version 1.68.1
> Found Python 3 [Python.Python.3] Version 3.10.5150.0
> Found Tiled [Tiled.Tiled] Version 1.8.5
> Found Epic Games Launcher [EpicGames.EpicGamesLauncher] Version 1.3.23.0
> Found OpenVPN [OpenVPNTechnologies.OpenVPN] Version 2.5.036
> Found Unity Hub [UnityTechnologies.UnityHub] Version 3.1.2
> Found Viber [Viber.Viber] Version 17.8.1.0
> Found Waves Central [WavesAudio.WavesCentral] Version 13.1.4
> (1/21) Found 7-Zip [7zip.7zip] Version 22.00
> This application is licensed to you by its owner.
> Microsoft is not responsible for, nor does it grant any licenses to, third-party packages.
> Downloading https://www.7-zip.org/a/7z2200-x64.exe
>   ██████████████████████████████  1.49 MB / 1.49 MB
> Successfully verified installer hash
> Starting package install...
> Successfully installed
> 
> (2/21) Found Driver Booster 9 [IObit.DriverBooster9] Version 9.3.0
> This application is licensed to you by its owner.
> Microsoft is not responsible for, nor does it grant any licenses to, third-party packages.
> Downloading https://cdn.iobit.com/dl/driver_booster_setup.exe
>   ██████████████████████████████  24.5 MB / 24.5 MB
> Installer hash does not match; this cannot be overridden when running as admin
> 
> (3/21) Found Product Portal [iZotope.ProductPortal] Version 1.4.3
> This application is licensed to you by its owner.
> Microsoft is not responsible for, nor does it grant any licenses to, third-party packages.
> Downloading https://s3.amazonaws.com/izotopedownloads/product_download/iZotope_Product_Portal_v1_4_3.exe
>   ██████████████████████████████   113 MB /  113 MB
> Successfully verified installer hash
> Starting package install...
> Successfully installed
> 
> (4/21) Found Unity [UnityTechnologies.Unity.2020] Version 2020.3.26f1
> This application is licensed to you by its owner.
> Microsoft is not responsible for, nor does it grant any licenses to, third-party packages.
> Downloading https://download.unity3d.com/download_unity/7298b473bc1a/Windows64EditorInstaller/UnitySetup64-2020.3.26f1.exe
> Package download canceled.
> 
> Cancelled


> C:\WINDOWS\system32>winget upgrade --all -h
> Found Driver Booster 9 [IObit.DriverBooster9] Version 9.3.0
> Found Product Portal [iZotope.ProductPortal] Version 1.4.3
> Found Unity [UnityTechnologies.Unity.2020] Version 2020.3.26f1
> Found WhySoSlow [Resplendence.WhySoSlow] Version 1.61
> Found Microsoft Visual C++ 2013 Redistributable (x64) [Microsoft.VC++2013Redist-x64] Version 12.0.40664.0
> Found Eclipse Temurin JDK with Hotspot 8 [EclipseAdoptium.Temurin.8] Version 8.0.332.9
> Found LibreOffice [TheDocumentFoundation.LibreOffice] Version 7.3.4.2
> Found Microsoft XNA Framework Redistributable Refresh [Microsoft.XNARedist] Version 4.0.30901.0
> Found Microsoft Visual C++ 2015-2022 Redistributable (x86) [Microsoft.VC++2015-2022Redist-x86] Version 14.32.31326.0
> Found Microsoft Visual C++ 2013 Redistributable (x86) [Microsoft.VC++2013Redist-x86] Version 12.0.40664.0
> Found Poedit [VaclavSlavik.Poedit] Version 3.0
> Found GOG GALAXY [GOG.Galaxy] Version 2.0.49.1
> Found Microsoft Visual Studio Code [Microsoft.VisualStudioCode] Version 1.68.1
> Found Python 3 [Python.Python.3] Version 3.10.5150.0
> Found Tiled [Tiled.Tiled] Version 1.8.5
> Found Epic Games Launcher [EpicGames.EpicGamesLauncher] Version 1.3.23.0
> Found OpenVPN [OpenVPNTechnologies.OpenVPN] Version 2.5.036
> Found Unity Hub [UnityTechnologies.UnityHub] Version 3.1.2
> Found Viber [Viber.Viber] Version 17.8.1.0
> Found Waves Central [WavesAudio.WavesCentral] Version 13.1.4
> (1/20) Found Driver Booster 9 [IObit.DriverBooster9] Version 9.3.0
> This application is licensed to you by its owner.
> Microsoft is not responsible for, nor does it grant any licenses to, third-party packages.
> Downloading https://cdn.iobit.com/dl/driver_booster_setup.exe
>   ██████████████████████████████  24.5 MB / 24.5 MB
> Installer hash does not match; this cannot be overridden when running as admin
> 
> (2/20) Found Product Portal [iZotope.ProductPortal] Version 1.4.3
> This application is licensed to you by its owner.
> Microsoft is not responsible for, nor does it grant any licenses to, third-party packages.
> Downloading https://s3.amazonaws.com/izotopedownloads/product_download/iZotope_Product_Portal_v1_4_3.exe
>   ██████████████████████████████   113 MB /  113 MB
> Successfully verified installer hash
> Starting package install...
> Successfully installed
> 
> (3/20) Found Unity [UnityTechnologies.Unity.2020] Version 2020.3.26f1
> This application is licensed to you by its owner.
> Microsoft is not responsible for, nor does it grant any licenses to, third-party packages.
> Downloading https://download.unity3d.com/download_unity/7298b473bc1a/Windows64EditorInstaller/UnitySetup64-2020.3.26f1.exe
> Package download canceled.
> 
> Cancelled
